File: gnsdk-3.08.6.5437o-20170914/wrappers/gnsdk_java/src_wrapper/j2se/com/gracenote/gnsdk/GnRhythmStation.java

package com.gracenote.gnsdk;

/** 
** Provides services for interacting with Rhythm Stations 
*/ 
 
public class GnRhythmStation {
  private transient long swigCPtr;
  protected transient boolean swigCMemOwn;

  protected GnRhythmStation(long cPtr, boolean cMemoryOwn) {
    swigCMemOwn = cMemoryOwn;
    swigCPtr = cPtr;
  }

  protected static long getCPtr(GnRhythmStation obj) {
    return (obj == null) ? 0 : obj.swigCPtr;
  }

  protected void finalize() {
    delete();
  }

  public synchronized void delete() {
    if (swigCPtr != 0) {
      if (swigCMemOwn) {
        swigCMemOwn = false;
        gnsdk_javaJNI.delete_GnRhythmStation(swigCPtr);
      }
      swigCPtr = 0;
    }
  }

	private IGnStatusEvents pEventHandler;
	private IGnStatusEventsProxyU eventHandlerProxy;

/** 
*  Constructs a Rhythm Station object from a {@link GnRhythmQuery} object and event delegate 
*  @param rhythmQuery   [in] {@link GnRhythmQuery} object containing the seeds and options to generate a new Rhythm Station. 
*  @param filter		 [in] {@link GnRhythmFilter} object containing the filter to generate a new Rhythm Station. 
*  @param pEventHandler [in-opt] Set Optional status event handler to get bytes sent, received, or completed. 
*/ 
 
  public GnRhythmStation(GnRhythmQuery rhythmQuery, GnRhythmFilter filter, IGnStatusEvents pEventHandler) throws com.gracenote.gnsdk.GnException {
	this(0, true);
	
	eventHandlerProxy = new IGnStatusEventsProxyU(pEventHandler);
	this.pEventHandler=pEventHandler;	// <REFERENCE_NAME_CHECK><TYPE>IGnStatusEvents</TYPE><NAME>pEventHandler</NAME></REFERENCE_NAME_CHECK> leave for scripted verification of names
	
    swigCPtr = gnsdk_javaJNI.new_GnRhythmStation__SWIG_0(GnRhythmQuery.getCPtr(rhythmQuery), rhythmQuery, GnRhythmFilter.getCPtr(filter), filter, (eventHandlerProxy==null)?0:IGnStatusEventsProxyL.getCPtr(eventHandlerProxy), eventHandlerProxy);
}

  public GnRhythmStation(GnRhythmQuery rhythmQuery, GnRhythmFilter filter) throws com.gracenote.gnsdk.GnException {
	this(0, true);
	
	eventHandlerProxy = new IGnStatusEventsProxyU(pEventHandler);
	this.pEventHandler=pEventHandler;	// <REFERENCE_NAME_CHECK><TYPE>IGnStatusEvents</TYPE><NAME>pEventHandler</NAME></REFERENCE_NAME_CHECK> leave for scripted verification of names
	
    swigCPtr = gnsdk_javaJNI.new_GnRhythmStation__SWIG_1(GnRhythmQuery.getCPtr(rhythmQuery), rhythmQuery, GnRhythmFilter.getCPtr(filter), filter);
}

  public GnRhythmStation(GnRhythmQuery rhythmQuery, IGnStatusEvents pEventHandler) throws com.gracenote.gnsdk.GnException {
	this(0, true);
	
	eventHandlerProxy = new IGnStatusEventsProxyU(pEventHandler);
	this.pEventHandler=pEventHandler;	// <REFERENCE_NAME_CHECK><TYPE>IGnStatusEvents</TYPE><NAME>pEventHandler</NAME></REFERENCE_NAME_CHECK> leave for scripted verification of names
	
    swigCPtr = gnsdk_javaJNI.new_GnRhythmStation__SWIG_2(GnRhythmQuery.getCPtr(rhythmQuery), rhythmQuery, (eventHandlerProxy==null)?0:IGnStatusEventsProxyL.getCPtr(eventHandlerProxy), eventHandlerProxy);
}

  public GnRhythmStation(GnRhythmQuery rhythmQuery) throws com.gracenote.gnsdk.GnException {
	this(0, true);
	
	eventHandlerProxy = new IGnStatusEventsProxyU(pEventHandler);
	this.pEventHandler=pEventHandler;	// <REFERENCE_NAME_CHECK><TYPE>IGnStatusEvents</TYPE><NAME>pEventHandler</NAME></REFERENCE_NAME_CHECK> leave for scripted verification of names
	
    swigCPtr = gnsdk_javaJNI.new_GnRhythmStation__SWIG_3(GnRhythmQuery.getCPtr(rhythmQuery), rhythmQuery);
}

/** 
*  Constructs a Rhythm Station object from an existing Rhythm Station ID, with a Gracenote user and event delegate 
*  @param stationId    [in] An Rhythm Station ID which will be used to retrieve a pre-existing Rhythm Station. 
*  @param user          [in] Set {@link GnUser} object representing the user making the {@link GnMusicId} request. 
*  @param pEventHandler [in-opt] Set Optional status event handler to get bytes sent, received, or completed. 
*/ 
 
  public GnRhythmStation(String stationId, GnUser user, IGnStatusEvents pEventHandler) throws com.gracenote.gnsdk.GnException {
	this(0, true);
	
	eventHandlerProxy = new IGnStatusEventsProxyU(pEventHandler);
	this.pEventHandler=pEventHandler;	// <REFERENCE_NAME_CHECK><TYPE>IGnStatusEvents</TYPE><NAME>pEventHandler</NAME></REFERENCE_NAME_CHECK> leave for scripted verification of names
	
    swigCPtr = gnsdk_javaJNI.new_GnRhythmStation__SWIG_4(stationId, GnUser.getCPtr(user), user, (eventHandlerProxy==null)?0:IGnStatusEventsProxyL.getCPtr(eventHandlerProxy), eventHandlerProxy);
}

  public GnRhythmStation(String stationId, GnUser user) throws com.gracenote.gnsdk.GnException {
	this(0, true);
	
	eventHandlerProxy = new IGnStatusEventsProxyU(pEventHandler);
	this.pEventHandler=pEventHandler;	// <REFERENCE_NAME_CHECK><TYPE>IGnStatusEvents</TYPE><NAME>pEventHandler</NAME></REFERENCE_NAME_CHECK> leave for scripted verification of names
	
    swigCPtr = gnsdk_javaJNI.new_GnRhythmStation__SWIG_5(stationId, GnUser.getCPtr(user), user);
}

/** 
*  Retrieves the station ID 
*   @return A C String contaning the Rhythm Station's ID. 
*  <p><b>Remarks:</b></p> 
*  The returned string is a constant. Do not attempt to modify or delete. 
*/ 
 
  public String stationId() throws com.gracenote.gnsdk.GnException {
    return gnsdk_javaJNI.GnRhythmStation_stationId(swigCPtr, this);
  }

/** 
*  Generate a Rhythm Station Playlist. 
*  @return An instance of {@link GnResponseAlbums}, response contains one Album per Recommended Track. 
*  <p><b>Remarks:</b></p> 
*  The Matched Track on each Album in the {@link GnResponseAlbums} is the Recommended Track 
*/ 
 
  public GnResponseAlbums generatePlaylist() throws com.gracenote.gnsdk.GnException {
    return new GnResponseAlbums(gnsdk_javaJNI.GnRhythmStation_generatePlaylist(swigCPtr, this), true);
  }

/** 
*  Adds an event to a Rhythm station. 
*   @param event        [in] One of the predefined Rhythm event types 
*   @param gnObj        [in] A {@link GnDataObject} representing the context of the event 
*/ 
 
  public void event(GnRhythmEvent event, GnDataObject gnObj) throws com.gracenote.gnsdk.GnException {
    gnsdk_javaJNI.GnRhythmStation_event(swigCPtr, this, event.swigValue(), GnDataObject.getCPtr(gnObj), gnObj);
  }

/** 
* Get the event handler provided on construction 
* @return Event handler 
*/ 
 
  public IGnStatusEvents eventHandler() {
	return pEventHandler;
}

/** 
* Get {@link GnRhythmQuery} options object. Use to configure your {@link GnRhythmQuery} instance. 
* @return Options object 
*/ 
 
  public GnRhythmQueryOptions options() {
    return new GnRhythmQueryOptions(gnsdk_javaJNI.GnRhythmStation_options(swigCPtr, this), false);
  }

/** 
* Set cancel state 
* @param bCancel   [in] Cancel state 
*/ 
 
  public void setCancel(boolean bCancel) {
    gnsdk_javaJNI.GnRhythmStation_setCancel(swigCPtr, this, bCancel);
  }

/** 
* Get cancel state. 
* @return Cancel state 
*/ 
 
  public boolean isCancelled() {
    return gnsdk_javaJNI.GnRhythmStation_isCancelled(swigCPtr, this);
  }

}

File: gnsdk-3.08.6.5437o-20170914/wrappers/gnsdk_csharp/src_wrapper/GnRhythmStation.cs

namespace GracenoteSDK {

/**
* \class GnRhythmStation
* Provides services for interacting with Rhythm Stations
*/
public class GnRhythmStation : global::System.IDisposable {
  private global::System.Runtime.InteropServices.HandleRef swigCPtr;
  protected bool swigCMemOwn;

  internal GnRhythmStation(global::System.IntPtr cPtr, bool cMemoryOwn) {
    swigCMemOwn = cMemoryOwn;
    swigCPtr = new global::System.Runtime.InteropServices.HandleRef(this, cPtr);
  }

  internal static global::System.Runtime.InteropServices.HandleRef getCPtr(GnRhythmStation obj) {
    return (obj == null) ? new global::System.Runtime.InteropServices.HandleRef(null, global::System.IntPtr.Zero) : obj.swigCPtr;
  }

  ~GnRhythmStation() {
    Dispose();
  }

  public virtual void Dispose() {
    lock(this) {
      if (swigCPtr.Handle != global::System.IntPtr.Zero) {
        if (swigCMemOwn) {
          swigCMemOwn = false;
          gnsdk_csharp_marshalPINVOKE.delete_GnRhythmStation(swigCPtr);
        }
        swigCPtr = new global::System.Runtime.InteropServices.HandleRef(null, global::System.IntPtr.Zero);
      }
      global::System.GC.SuppressFinalize(this);
    }
  }

/**
*  Constructs a Rhythm Station object from a GnRhythmQuery object and event delegate
*  @param rhythmQuery   [in] GnRhythmQuery object containing the seeds and options to generate a new Rhythm Station.
*  @param filter		 [in] GnRhythmFilter object containing the filter to generate a new Rhythm Station.
*  @param pEventHandler [in-opt] Set Optional status event handler to get bytes sent, received, or completed.
*/
  public GnRhythmStation(GnRhythmQuery rhythmQuery, GnRhythmFilter filter, GnStatusEventsDelegate pEventHandler) : this(gnsdk_csharp_marshalPINVOKE.new_GnRhythmStation__SWIG_0(GnRhythmQuery.getCPtr(rhythmQuery), GnRhythmFilter.getCPtr(filter), GnStatusEventsDelegate.getCPtr(pEventHandler)), true) {
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

  public GnRhythmStation(GnRhythmQuery rhythmQuery, GnRhythmFilter filter) : this(gnsdk_csharp_marshalPINVOKE.new_GnRhythmStation__SWIG_1(GnRhythmQuery.getCPtr(rhythmQuery), GnRhythmFilter.getCPtr(filter)), true) {
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

  public GnRhythmStation(GnRhythmQuery rhythmQuery, GnStatusEventsDelegate pEventHandler) : this(gnsdk_csharp_marshalPINVOKE.new_GnRhythmStation__SWIG_2(GnRhythmQuery.getCPtr(rhythmQuery), GnStatusEventsDelegate.getCPtr(pEventHandler)), true) {
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

  public GnRhythmStation(GnRhythmQuery rhythmQuery) : this(gnsdk_csharp_marshalPINVOKE.new_GnRhythmStation__SWIG_3(GnRhythmQuery.getCPtr(rhythmQuery)), true) {
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

/**
*  Constructs a Rhythm Station object from an existing Rhythm Station ID, with a Gracenote user and event delegate
*  @param stationId    [in] An Rhythm Station ID which will be used to retrieve a pre-existing Rhythm Station.
*  @param user          [in] Set GnUser object representing the user making the GnMusicId request.
*  @param pEventHandler [in-opt] Set Optional status event handler to get bytes sent, received, or completed.
*/
  public GnRhythmStation(string stationId, GnUser user, GnStatusEventsDelegate pEventHandler) : this(gnsdk_csharp_marshalPINVOKE.new_GnRhythmStation__SWIG_4(stationId, GnUser.getCPtr(user), GnStatusEventsDelegate.getCPtr(pEventHandler)), true) {
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

  public GnRhythmStation(string stationId, GnUser user) : this(gnsdk_csharp_marshalPINVOKE.new_GnRhythmStation__SWIG_5(stationId, GnUser.getCPtr(user)), true) {
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

/**
*  Retrieves the station ID
*   @return A C String contaning the Rhythm Station's ID.
*  <p><b>Remarks:</b></p>
*  The returned string is a constant. Do not attempt to modify or delete.
*/
  public string StationId() {
    string ret = gnsdk_csharp_marshalPINVOKE.GnRhythmStation_StationId(swigCPtr);
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
    return ret;
  }

/**
*  Generate a Rhythm Station Playlist.
*  @return An instance of GnResponseAlbums, response contains one Album per Recommended Track.
*  <p><b>Remarks:</b></p>
*  The Matched Track on each Album in the GnResponseAlbums is the Recommended Track
*/
  public GnResponseAlbums GeneratePlaylist() {
    GnResponseAlbums ret = new GnResponseAlbums(gnsdk_csharp_marshalPINVOKE.GnRhythmStation_GeneratePlaylist(swigCPtr), true);
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
    return ret;
  }

/**
*  Adds an event to a Rhythm station.
*   @param event        [in] One of the predefined Rhythm event types
*   @param gnObj        [in] A GnDataObject representing the context of the event
*/
  public void Event(GnRhythmEvent arg0, GnDataObject gnObj) {
    gnsdk_csharp_marshalPINVOKE.GnRhythmStation_Event(swigCPtr, (int)arg0, GnDataObject.getCPtr(gnObj));
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

/**
* Get the event handler provided on construction
* @return Event handler
*/
  public GnStatusEventsDelegate EventHandler() {
    global::System.IntPtr cPtr = gnsdk_csharp_marshalPINVOKE.GnRhythmStation_EventHandler(swigCPtr);
    GnStatusEventsDelegate ret = (cPtr == global::System.IntPtr.Zero) ? null : new GnStatusEventsDelegate(cPtr, false);
    return ret;
  }

/**
* Get GnRhythmQuery options object. Use to configure your GnRhythmQuery instance.
* @return Options object
*/
  public GnRhythmQueryOptions Options() {
    GnRhythmQueryOptions ret = new GnRhythmQueryOptions(gnsdk_csharp_marshalPINVOKE.GnRhythmStation_Options(swigCPtr), false);
    return ret;
  }

/**
* Set cancel state
* @param bCancel   [in] Cancel state
*/
  public virtual void SetCancel(bool bCancel) {
    gnsdk_csharp_marshalPINVOKE.GnRhythmStation_SetCancel(swigCPtr, bCancel);
  }

/**
* Get cancel state.
* @return Cancel state
*/
  public virtual bool IsCancelled() {
    bool ret = gnsdk_csharp_marshalPINVOKE.GnRhythmStation_IsCancelled(swigCPtr);
    return ret;
  }

}

}